tests/test_consumer/test_extractors/test_extractors_column_lineage.py

```diff
@@ -0,0 +1,354 @@
+                    OpenLineageColumnLineageDatasetFacetFieldRef(
+                        namespace="postgres://192.168.1.1:5432",
+                        name="mydb.myschema.mytable",
+                        field="source_col",
+                        transformations=[
+                            OpenLineageColumnLineageDatasetFacetFieldTransformation(
+                                type="DIRECT",
+                                subtype="TRANSFORMATION",
+                            ),
+                        ],
+                    ),
+                    OpenLineageColumnLineageDatasetFacetFieldRef(
+                        namespace="postgres://192.168.1.1:5432",
+                        name="mydb.myschema.mytable",
+                        field="source_col",
+                        transformations=[
+                            OpenLineageColumnLineageDatasetFacetFieldTransformation(
+                                type="DIRECT",
+                                subtype="AGGREGATION",
+                            ),
+                        ],
+                    ),
+                ],
+            ),
+        },
+    )
+    first_operation_id = UUID("0194fae4-6f09-7feb-9285-cea346ec4c14")
+    second_operation_id = UUID("0194fae4-e164-7a88-9afb-1d62d7daac95")
+    first_event = get_run_event_with_column_lineage(
+        first_operation_id,
+        column_lineage_first_facet,
+    )
+    second_event = get_run_event_with_column_lineage(
+        second_operation_id,
+        column_lineage_second_facet,
+    )
+    extracted = extract_batch([first_event, second_event])
+    column_lineage = extracted.column_lineage()
+
+    assert column_lineage == [
+        ColumnLineageDTO(
+            operation=get_operation_dto(first_operation_id),
+            source_dataset=extracted_postgres_dataset,
+            target_dataset=extracted_hive_dataset,
+            dataset_column_relations=[
+                DatasetColumnRelationDTO(
+                    type=DatasetColumnRelationTypeDTO.AGGREGATION,
+                    source_column="source_col",
+                    target_column="column",
+                    fingerprint=None,
+                ),
+            ],
+        ),
+        ColumnLineageDTO(
+            operation=get_operation_dto(second_operation_id),
+            source_dataset=extracted_postgres_dataset,
+            target_dataset=extracted_hive_dataset,
+            dataset_column_relations=[
+                DatasetColumnRelationDTO(
+                    type=DatasetColumnRelationTypeDTO.AGGREGATION,
+                    source_column="source_col",
+                    target_column="column",
+                    fingerprint=None,
+                ),
+                DatasetColumnRelationDTO(
+                    type=DatasetColumnRelationTypeDTO.TRANSFORMATION,
+                    source_column="source_col",
+                    target_column="column",
+                    fingerprint=None,
+                ),
```

Comment:
If source & target columns are the same, then there should be only one item with `type=DatasetColumnRelationTypeDTO.AGGREGATION | DatasetColumnRelationTypeDTO.TRANSFORMATION`

Reply:
You mean that this test-case is imposible in reality or it's wrong behavior?

Reply:
Under the hood, ColumnLineageDTO merges these 2 items into one:
https://github.com/MobileTeleSystems/data-rentgen/blob/06a3124347bafbb0adda79f89333529714efc704/data_rentgen/dto/column_lineage.py#L31-L32

Result is the same, but current test case should be more descriptive.

Reply:
In test configuration like in the last commit should two transformations be merged into one?

Reply:
Only if both transformations have the same combination of source_column + target_column